### DISCUSSION TITLE
Inquiry about Changes in go-netbox Version 3.7.1 Alpha

### DISCUSSION THREAD

**Discussion by icetimux** [netbox-community/go-netbox#167] [General]
Hello go-netbox Developers,

I hope this message finds you well. I recently noticed that in the 3.7.1 alpha version of go-netbox, the netbox/dcim directory seems to be absent. In our project, we heavily rely on the dcim module, importing it like this:

```
import (
	"github.com/netbox-community/go-netbox/netbox/client/dcim"
)

```
Specifically, we use components such as *dcim.DcimPowerFeedsListParams. However, in the latest version (3.7.1 alpha), the dcim_power_feeds_list_parameters.go file is no longer present.

Could you kindly clarify if this directory has been deprecated or if it will be added back in future releases? Additionally, if there are changes in how we should import and use components like DcimPowerFeedsListParams in version 3.7.1, could you provide guidance on that?

Furthermore, do you plan to release a migration guide or any documentation that can assist users in updating from version 3.4.5 to version 3.7.1?

Your guidance on these matters would be greatly appreciated. Thank you for your time and assistance.

**Comment by jqueuniet** [comment]
Following the upstream Netbox switch from OpenAPI v2 to v3, this module had to be fully rewritten using a new code generator. The new one has a flat hierarchy and does not use sub packages anymore, but you will find all relevant objects at the root.

Ie, for power feeds: https://github.com/netbox-community/go-netbox/blob/master/model_power_feed.go
